src/compiler/checker.ts

```diff
@@ -9595,13 +9595,21 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     if (p.flags & SymbolFlags.Accessor && useAccessors) {
                         const result: AccessorDeclaration[] = [];
                         if (p.flags & SymbolFlags.SetAccessor) {
+                            const setter = p.declarations?.find((d): d is SignatureDeclaration => {

```

Comment:
If a setter has an overloaded signature, what happens here?

Reply:
I'm not even sure how to write such a thing - but perhaps it's possible somehow when the setter is defined using `Object.defineProperty` call in JS file? Do you have an example of an overloaded setter at hand?

Reply:
Yeah, you're right, there's no setter that could be written that could make use of any extra signatures, since you only have one parameter and no return anyway.